### ISSUE TITLE
Logical incompatibilities between MONDO and HPO

### ISSUE THREAD

**Issue by balhoff** [monarch-initiative/mondo#356]
I'd like to use MONDO and HPO together with a reasoner. This doesn't seem to be possible at the moment. Here's an example:

## [Adenocarcinoma of the colon](http://purl.obolibrary.org/obo/HP_0040276) EquivalentTo [Nothing](http://www.w3.org/2002/07/owl#Nothing) ##

  - [Adenocarcinoma of the colon](http://purl.obolibrary.org/obo/HP_0040276) EquivalentTo [has part](http://purl.obolibrary.org/obo/BFO_0000051) some 
([increased amount](http://purl.obolibrary.org/obo/PATO_0000470) and ([inheres in](http://purl.obolibrary.org/obo/RO_0000052) some 
([Adenocarcinoma](http://purl.obolibrary.org/obo/NCIT_C2852) and ([part of](http://purl.obolibrary.org/obo/BFO_0000050) some [colon](http://purl.obolibrary.org/obo/UBERON_0001155)))) and ([has modifier](http://purl.obolibrary.org/obo/RO_0002573) some [abnormal](http://purl.obolibrary.org/obo/PATO_0000460)))
    - [Adenocarcinoma](http://purl.obolibrary.org/obo/NCIT_C2852) SubClassOf [Carcinoma](http://purl.obolibrary.org/obo/NCIT_C2916)
      - [carcinoma](http://purl.obolibrary.org/obo/MONDO_0004993) EquivalentTo [Carcinoma](http://purl.obolibrary.org/obo/NCIT_C2916)
        - [carcinoma](http://purl.obolibrary.org/obo/MONDO_0004993) EquivalentTo [cancer](http://purl.obolibrary.org/obo/MONDO_0004992) and ([disease arises from structure](http://purl.obolibrary.org/obo/RO_0004030) some [epithelial cell](http://purl.obolibrary.org/obo/CL_0000066))
          - [disease arises from structure](http://purl.obolibrary.org/obo/RO_0004030) SubPropertyOf [disease has basis in](http://purl.obolibrary.org/obo/RO_0004019)
            - [disease has basis in](http://purl.obolibrary.org/obo/RO_0004019) SubPropertyOf [causal relationship with disease as subject](http://purl.obolibrary.org/obo/RO_0004023)
              - [causal relationship with disease as subject](http://purl.obolibrary.org/obo/RO_0004023) Domain [disposition](http://purl.obolibrary.org/obo/BFO_0000016)
                - [disposition](http://purl.obolibrary.org/obo/BFO_0000016) SubClassOf [realizable entity](http://purl.obolibrary.org/obo/BFO_0000017)
                  - [realizable entity](http://purl.obolibrary.org/obo/BFO_0000017) SubClassOf [specifically dependent continuant](http://purl.obolibrary.org/obo/BFO_0000020)
    - [colon](http://purl.obolibrary.org/obo/UBERON_0001155) SubClassOf [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [large intestine](http://purl.obolibrary.org/obo/UBERON_0000059)
      - [large intestine](http://purl.obolibrary.org/obo/UBERON_0000059) SubClassOf [part of](http://purl.obolibrary.org/obo/BFO_0000050) some [lower digestive tract](http://purl.obolibrary.org/obo/UBERON_0004907)
        - [lower digestive tract](http://purl.obolibrary.org/obo/UBERON_0004907) SubClassOf [develops from](http://purl.obolibrary.org/obo/RO_0002202) some [hindgut](http://purl.obolibrary.org/obo/UBERON_0001046)
          - [develops from](http://purl.obolibrary.org/obo/RO_0002202) Domain [independent continuant](http://purl.obolibrary.org/obo/BFO_0000004)
  - [part of](http://purl.obolibrary.org/obo/BFO_0000050) o [develops from](http://purl.obolibrary.org/obo/RO_0002202) SubPropertyOf [develops from](http://purl.obolibrary.org/obo/RO_0002202)
  - [independent continuant](http://purl.obolibrary.org/obo/BFO_0000004) DisjointWith [specifically dependent continuant](http://purl.obolibrary.org/obo/BFO_0000020)

Not sure if this is more a problem of HPO or of MONDO. But HPO is treating carcinoma as a structure, and MONDO is treating it as a disease. Perhaps HPO just needs a different relation to connect carcinoma to a location.

cc @cmungall @pnrobinson

**Comment by pnrobinson** [comment]
The HPO and the MONDO terms for an adenocarcinoma are not referring to the same thing, and so the relation shoudl be


MONDO ----has_phenotype--->HPO


even though the term labels may be the same or similar!

-Peter


Peter Robinson
Professor and Donald A. Roux Chair, Genomics and Computational Biology
The Jackson Laboratory for Genomic Medicine
860.837.2095 t | peter.robinson@jax.org | https://robinsongroup.github.io/
Peter Robinson
________________________________

**Comment by balhoff** [comment]
I see that, however HPO also uses in its logical definition the NCIT Adenocarcinoma, saying it's part of the colon. And MONDO is equating NCIT Carcinoma (superclass of NCIT Adenocarcinoma) to MONDO carcinoma. So maybe the primary issue is that HPO and MONDO are taking different interpretations of what NCIT is representing.

**Comment by cmungall** [comment]
Correct, HPO is effectively injecting an inference that 
NCIT:Adenocarcinoma (and other neoplastic diseases) is a material 
entity.

This is neither right nor wrong, because NCIT undercommits to BFO

  / NCIT:C7057 ! Disease, Disorder or Finding
   is_a NCIT:C2991 ! Disease or Disorder
    is_a NCIT:C3262 ! Neoplasm
     is_a NCIT:C4741 ! Neoplasm by Morphology
      is_a NCIT:C3709 ! Epithelial Neoplasm
       is_a NCIT:C2916 ! Carcinoma
        is_a NCIT:C2852 ! Adenocarcinoma ***

"disease or disorder" is intentionally left vague.

Similarly, MONDO effectively injects an inference that 
NCIT:Adenocarcinoma is a disposition/state, which is disjoint from 
material entity.

We really need to resolve this.

As an aside, when users see things like "Adenocarcinoma has_phenotype 
adenocarcinoma" they get justifiably confused. At the very least we need 
consistent labeling to indicate the different meanings of things. I have 
been suffixing "(disease)" on labels in MONDO that are duplicative of 
HPO, but this is not ideal.

Possible solutions (not all mutually exclusive):

1. Treat NCIT 'neoplasm' and subclasses as MEs. This would have 
implications for MONDO since we either would have a different kind of 
axiom relating NCIT to MONDO, or the ontological commitment would 
propagate to MONDO

2. Ban nested post-composition with NCIT in HPO. There is no need for 
it, NCIT is already heavily pre-composed, it has adenocarcinoma of colon 
already

3. Use MONDO DPs in nested post-compositions with NCIT in HPO (e.g. 
disease_has_phenotype). This is decidedly odd here as you are using a 
disease to define a phenotype.

4. Create a new ontology that is explicitly ME, shadowing NCIT, and use 
that in HPO logical defs. Users will be delighted with this [sarcasm], 
as we will have HPO:adenocarcinoma, MONDO:adenocarcinoma, 
NCIT:adenocarcinoma, PathoMaterialEntity:adenocarcinoma

5. Remove all neoplasm terms from HPO, and allow use of either NCIT or 
MONDO in annotation

6. Follow NCIT and explicitly defer commitment, perhaps using Stefan S's 
unionOf strategy. This has many issues though.

[btw, love the markdown explanation, can we hook this into ROBOT?]

**Comment by pnrobinson** [comment]
Hi Chris,

consider that for HPO,

Adenocarcinoma is the actual tumor, i.e., a material entity. For MONDO,

Adenocarcinoma is a disease characterized by this tumor as well as potentially other manifestations such as weight loss and pain, and by an etiology (mutations), a time course (progression through stages) and a response to treatment. This can reasonably be modeled as a disposition.

I would argue that the current state of modelling is at least a good start on what we need to do!

-petert


Peter Robinson
Professor and Donald A. Roux Chair, Genomics and Computational Biology
The Jackson Laboratory for Genomic Medicine
860.837.2095 t | peter.robinson@jax.org | https://robinsongroup.github.io/
Peter Robinson
________________________________

**Comment by balhoff** [comment]
> [btw, love the markdown explanation, can we hook this into ROBOT?]

`robot explain`: https://github.com/ontodev/robot/pull/354

I need to tweak the options and get it merged!

**Comment by cmungall** [comment]
See #461 for a potential solution

**Comment by nicolevasilevsky** [comment]
@balhoff can this be closed?

**Comment by balhoff** [comment]
Yes! Excellent work. :-)
